updates for pe v3.8.* and 2015.2

abuxton opened the pull request.

abuxton committed:
create node_manager as a class to deploy the puppet-classifier gem

init, to install

install class

update to include install

examples added

update string enc variables

missing parenthasis

variabalised noop

remove erroneus ,

init retrospec

stanza issue on osx10.5.5 so commented out

refactor for puppet4 reserved words now inc noop

refactor for puppet4 reserved words now inc noop

respawned spec

respawned spec

create gitignore (.lock)

refactor for params, and class test

test to ensure require

WhatsARanjit commented:
@abuxton Not to dole out more work, but can you add the corresponding spec tests?

WhatsARanjit commented:
@abuxton Also, this module does work on 3.7.x with the NC API.  The puppet_gem provider wasn't introduced yet.

abuxton commented:
hey Ranjit, i've now added tests and squashed the commit as no one needs to see that mess :)
Not sure about your last comment, the class is not needed in 3.8 hence why it just is executed in noop mode if some one should classify it for any reason!  
#puppethack

WhatsARanjit commented on spec/shared_contexts.rb:
```diff
@@ -0,0 +1,36 @@
+
+shared_context :real_hiera_data do
+    let(:hiera_config) do
+       hirea_config_file
```
Typo here?

abuxton committed:
removed shared_contexts as notrequired

abuxton commented:
Yeah, retrospec file not required as no hieradata in the module, so I have removed it.

abuxton committed:
rebase

abuxton committed:
bad merge, my bad

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/puppetlabs/puppetlabs-node_manager?pullRequest=3) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/puppetlabs/puppetlabs-node_manager?pullRequest=3) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/puppetlabs/puppetlabs-node_manager?pullRequest=3) it.</sub>
